src/main/hook/patch/gfx.c

```diff
@@ -191,17 +280,60 @@ XVisualInfo *glXChooseVisual(Display *dpy, int screen, int *attribList)
   return res;
 }
 
+// Check for fullscreen draw calls and scale them appropriately.
+void glDrawPixels(GLsizei width, GLsizei height, GLenum format, GLenum type, const GLvoid *pixels) {
+  if (!patch_gfx_real_glDrawPixels) {
+    patch_gfx_real_glDrawPixels =
+        (glDrawPixels_t) cnh_lib_get_func_addr("glDrawPixels");
+  }
+    // If our texture is the original screen size, we'll zoom, flip it because reasons, and then draw
+    // it to our updated zoom factor.    
+    if (width == s_init_display_width && height == s_init_display_height) {
+        // Do this - I don't remember why but do it.        
+        glRasterPos2i(0,0);       
+        glPixelZoom(s_zoom_factor_x, s_zoom_factor_y);
+        
+        // Create a new buffer to store the flipped image data
+        GLubyte *flippedPixels = malloc(width * height * sizeof(GLubyte) * 3);
+
+        // Flip the image data
+        for (int i = 0; i < height; i++) {
+            GLubyte *srcLine = ((GLubyte *)pixels) + i * width * 3;
+            GLubyte *dstLine = flippedPixels + (height - i - 1) * width * 3;
+            memcpy(dstLine, srcLine, width * 3);
```

Comment:
Overall, this approach might even be more expensive regarding performance than just scaling the entire framebuffer because you also need to issue additional gl calls in the other scaling logic.
Likely doesn't matter to pump, but I don't see performance as a benefit here at least.